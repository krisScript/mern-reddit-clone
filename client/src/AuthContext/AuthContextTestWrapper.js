import {AuthContextData} from './AuthContext'
import {BrowserRouter} from 'react-router-dom'
import React from 'react'
const AuthContextTestWrapper = ({authState,children,updateUserDataReducer}) => {
    return (
       <AuthContextData.Provider value={{authState,updateUserDataReducer}}>
        <BrowserRouter>
          {children}
          </BrowserRouter>
        </AuthContextData.Provider>
    )
}
export default AuthContextTestWrapper 