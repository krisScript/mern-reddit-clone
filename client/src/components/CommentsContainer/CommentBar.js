import React,{useState,useContext} from 'react'
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import useAuthContext from '../../hooks/useAuthContext/useAuthContext'
import {
  faShare,faCommentAlt, faBookmark,faTrashAlt,faPen,
} from '@fortawesome/free-solid-svg-icons';
import deleteData from '../../util/deleteData'
library.add(faShare,faCommentAlt,faBookmark,faTrashAlt,faPen,);
const CommentBar= props => {
    const {userId,token} = useAuthContext
    const {authorId,comments,deleteCommentElement,commentId,toggle,setEditComment,comment,on} = props
    const deleteHandler = async () => {
        const apiUrl = `http://localhost:8080/comment/delete/${comment._id}`
        const data = await deleteData(apiUrl,token)
        if(data.msg === 'Comment Deleted'){
            deleteCommentElement(commentId)
        }
    }
    const editHandler = () => {
        if(on){

        }else {
            toggle()
            setEditComment(comment)
        }
    }
    let autorizedContent = ''
     if(authorId === userId){
        autorizedContent = 
        <>
         <button onClick={deleteHandler} className="button post-info-button">
        <FontAwesomeIcon icon="trash-alt" /><span>Delete</span>
        </button>
        <button onClick={editHandler} className="button post-info-button">
        <FontAwesomeIcon icon="pen" /><span>Edit</span>
        </button>
        </>
    }
    
    return (
        <div>
            <div className="post-info">
        <button className="button post-info-button">
        <FontAwesomeIcon icon="share" /><span>Share</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="comment-alt" /><span>{comments}</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="bookmark" /><span>Save</span>
        </button>
        {autorizedContent}
        </div>
        </div>
    )
}
export default CommentBar