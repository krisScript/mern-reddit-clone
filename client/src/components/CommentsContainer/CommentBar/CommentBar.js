import React,{useState,useContext} from 'react'
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import useAuthContext from '../../../hooks/useAuthContext/useAuthContext'
import {
  faShare,faCommentAlt, faBookmark,faTrashAlt,faPen,
} from '@fortawesome/free-solid-svg-icons';
import deleteData from '../../../util/deleteData'
library.add(faShare,faCommentAlt,faBookmark,faTrashAlt,faPen,);
const CommentBar= ({deleteCommentElement,toggle,setEditComment,comment,on}) => {
    const {userId,token} = useAuthContext()
    const {authorId,comments,_id} = comment
    const deleteHandler = async () => {
        const apiUrl = `http://localhost:8080/comment/delete/${_id}`
        const data = await deleteData(apiUrl,token)
        if(data.msg === 'deleated'){
            deleteCommentElement(_id)
        }
    }
    const editHandler = () => {
        if(on){

        }else {
            toggle()
            setEditComment(comment)
        }
    }
    let autorizedContent = ''
     if(authorId === userId){
        autorizedContent = 
        <>
         <button onClick={deleteHandler} className="button post-info-button" data-testid="delete-button">
        <FontAwesomeIcon icon="trash-alt" /><span>Delete</span>
        </button>
        <button onClick={editHandler} className="button post-info-button" data-testid="edit-button">
        <FontAwesomeIcon icon="pen" /><span>Edit</span>
        </button>
        </>
    }
    
    return (
        <div>
            <div className="post-info">
        <button className="button post-info-button" data-testid="share-button">
        <FontAwesomeIcon icon="share" /><span>Share</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="comment-alt" /><span>{comments}</span>
        </button>
        <button className="button post-info-button" data-testid="save-button">
        <FontAwesomeIcon icon="bookmark" /><span>Save</span>
        </button>
        {autorizedContent}
        </div>
        </div>
    )
}
export default CommentBar