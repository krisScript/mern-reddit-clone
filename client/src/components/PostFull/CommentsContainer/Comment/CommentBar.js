import React,{useState,useContext} from 'react'
import deleteComment from './deleteComment'
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { AuthContextData } from '../../../../AuthContext/AuthContext';
import {
  faShare,faCommentAlt, faBookmark,faTrashAlt,faPen,
} from '@fortawesome/free-solid-svg-icons';
library.add(faShare,faCommentAlt,faBookmark,faTrashAlt,faPen,);
const CommentBar= props => {
    const {authState} = useContext(AuthContextData)
    const {userId,token} = authState
    const {authorId,comments,deleteCommentElement,commentId,toggle,setEditComment,comment,on} = props
    const deleteHandler = async () => {
        console.log(authState)
        const data = await deleteComment(commentId,token)
        if(data.msg === 'Comment Deleted'){
            deleteCommentElement(commentId)
        }
    }
    const editHandler = () => {
        if(on){

        }else {
            toggle()
            setEditComment(comment)
        }
    }
    let autorizedContent = ''
     if(authorId === userId){
        autorizedContent = 
        <>
         <button onClick={deleteHandler} className="button post-info-button">
        <FontAwesomeIcon icon="trash-alt" /><span>Delete</span>
        </button>
        <button onClick={editHandler} className="button post-info-button">
        <FontAwesomeIcon icon="pen" /><span>Edit</span>
        </button>
        </>
    }
    
    return (
        <div>
            <div className="post-info">
        <button className="button post-info-button">
        <FontAwesomeIcon icon="share" /><span>Share</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="comment-alt" /><span>{comments}</span>
        </button>
        <button className="button post-info-button">
        <FontAwesomeIcon icon="bookmark" /><span>Save</span>
        </button>
        {autorizedContent}
        </div>
        </div>
    )
}
export default CommentBar