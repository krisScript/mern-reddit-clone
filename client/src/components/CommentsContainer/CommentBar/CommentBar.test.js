import React from 'react';
import { render, fireEvent, waitForElement,wait } from 'react-testing-library';
import CommentBar from './CommentBar';
import AuthContextTestWrapper from '../../../AuthContext/AuthContextTestWrapper'
const updateUserDataReducer = jest.fn()
jest.mock('../../../util/deleteData')
describe('<CommentBar/>',() => {
    const comment = {
        creationDate :'2019-04-01T13:51:12.885Z',
        communityId :'testId',
        authorId :"authorId",
        author :"Joseph Joestar",
        postId :'testPostId',
        postTitle: 'tstPostTitle',
        content :'Your next line is...',
        upvotes :10,
        downvotes :3,
        comments:1,
        _id:1,
    }
    const toggle = jest.fn()
    const deleteCommentElement  = jest.fn()
    const setEditComment  = jest.fn()
    const on = false
    const authState =  {isAuth:true,token:null,userId:comment.authorId,userData:null,updateUserDataReducer}
    const { container, rerender,queryByPlaceholderText,queryByValue ,getByTestId,getByText } = render (<CommentBar comment={comment} toggle={toggle} deleteCommentElement={deleteCommentElement} setEditComment={setEditComment} on={on} />,{wrapper: ({ children }) => (
      <AuthContextTestWrapper authState={authState} children={children} />
      )})
      it('snapshot',() => {
        expect(container).toMatchSnapshot()
    })

      it('should have edit button',  () => {
        const element =  getByTestId(`edit-button-${comment._id}`)
        expect(element).toBeTruthy;
        fireEvent.click(element)
        expect(toggle).toHaveBeenCalled()
        expect(setEditComment).toHaveBeenCalled()
        expect(setEditComment).toHaveBeenCalledWith(comment)
      });

      it('should have delete button',  () => {
        const element =  getByTestId(`delete-button`)
        expect(element).toBeTruthy;
        fireEvent.click(element)
        wait(() =>expect(deleteCommentElement).toHaveBeenCalled() )
        wait(() =>  expect(setEditComment).toHaveBeenCalledWith(comment._id) )
      
      }); 
       it('should have share button',  () => {
        const element =  getByTestId(`share-button`)
        expect(element).toBeTruthy;
      });

      it('should have save button',  () => {
        const element =  getByTestId(`save-button`)
        expect(element).toBeTruthy;
      });

      
      
    //   it('should have date element', async () => {
    //     const element =  getByText(humanReadableCreationDate)
    //     expect(element.className).toMatch('comment-date')
    //     expect(element).toBeTruthy;
    //   });
    //   it('should have content element', async () => {
    //     const element =  getByText(`${comment.content}`)
    //     expect(element.className).toMatch('comment-content')
    //     expect(element).toBeTruthy;
    //   });
      
})

