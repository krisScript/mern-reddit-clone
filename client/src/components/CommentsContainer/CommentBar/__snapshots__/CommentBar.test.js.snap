// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentBar/> snapshot 1`] = `
<div>
  <div>
    <div
      class="post-info"
    >
      <button
        class="button post-info-button"
        data-testid="share-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-share fa-w-16 "
          data-icon="share"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M503.691 189.836L327.687 37.851C312.281 24.546 288 35.347 288 56.015v80.053C127.371 137.907 0 170.1 0 322.326c0 61.441 39.581 122.309 83.333 154.132 13.653 9.931 33.111-2.533 28.077-18.631C66.066 312.814 132.917 274.316 288 272.085V360c0 20.7 24.3 31.453 39.687 18.164l176.004-152c11.071-9.562 11.086-26.753 0-36.328z"
            fill="currentColor"
          />
        </svg>
        <span>
          Share
        </span>
      </button>
      <button
        class="button post-info-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-alt fa-w-16 "
          data-icon="comment-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 9.8 11.2 15.5 19.1 9.7L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64z"
            fill="currentColor"
          />
        </svg>
        <span>
          1
        </span>
      </button>
      <button
        class="button post-info-button"
        data-testid="save-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bookmark fa-w-12 "
          data-icon="bookmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
            fill="currentColor"
          />
        </svg>
        <span>
          Save
        </span>
      </button>
      <button
        class="button post-info-button"
        data-testid="delete-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash-alt fa-w-14 "
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
          />
        </svg>
        <span>
          Delete
        </span>
      </button>
      <button
        class="button post-info-button"
        data-testid="edit-button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pen fa-w-16 "
          data-icon="pen"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
            fill="currentColor"
          />
        </svg>
        <span>
          Edit
        </span>
      </button>
    </div>
  </div>
</div>
`;
